linetools/spectra/xspectrum1d.py

```diff
@@ -276,46 +363,25 @@ def normalize(self, co=None, verbose=False, no_check=False):
         no_check: bool [False]
           Check size of array?
         """
-        # if already normalized give a warning and do nothing
-        try:
-            if self.normed is True:
-                print('The spectrum is already normalized! Aborting.')
-                return
-        except AttributeError: # self.normed is not defined
-            pass
-
-        # Sanity check
-        if co is None:
-            if hasattr(self, 'co') and self.co is not None:
-                co = self.co
-            else:
-                raise ValueError(
-                    'Must specify a continuum with `co` keyword.')
-        if (len(co) != len(self.flux)):
-            if no_check:
-                print('WARNING: Continuum length differs from flux')
-                if len(co) > len(self.flux):
-                    self.flux /= co[0:len(self.flux)]
-                    if self.uncertainty is not None:
-                        self.uncertainty.array /= co[0:len(self.flux)]
-                    return
+        if co is not None:
+            if (len(co) != self.npix):
+                if no_check:
```

Comment:
I have some innocuous SDSS analyzed data for QPQ.  I may
be able to get rid of this before too long but need it for now..